src/renderer/src/hooks/useMapeoDevice.ts

```diff
@@ -0,0 +1,25 @@
+import { useMemo } from 'react'
+import { mapeoDeviceStore, Role, useMapeoDeviceStoreAction } from './stores/mapeoDeviceStore'
+
+export const useMapeoDevice = (deviceId: string) => {
+  const device = mapeoDeviceStore(
+    (store) => store.devices[deviceId],
+    (oldVal, newVal) => JSON.stringify(oldVal) !== JSON.stringify(newVal),
```

Comment:
is this necessary? by default, `device` will change when `store.devices` or `deviceId` changes, which seems fine

Reply:
without this, anytime and device in the `device` object anything using this hook will get re-rendered. This makes it so that a re-render only happens when this specific device gets updated

Reply:
hm yeah i see. wondering if their suggested approach to this is what's described here: https://docs.pmnd.rs/zustand/recipes/recipes#memoizing-selectors

may not be addressing the same issue, but seems like it from a quick read

Reply:
I don't think it does? but maybe i don't think i fully understand it. the first approach memoizes based on id. so it will only rerender when id changes, not when the other properties change. So if we update the role, no rerender will happen.

The second approach i don't fully understand, so maybe it applies?

Reply:
Yeah the more I read it, the less sure I am that it's talking about the same problem. Certainly one of the odd details with Zustand 🤷 

Think it's more related to just preventing the selector function from being unnecessarily re-created on each render, which probably isn't a major issue in this case (doesn't hurt to apply their suggestion either).

Reply:
yeah, Im not to sure zustand is as robust as I thought. Maintaining complex lists in zustand seems to cause this problem. When objects are at the top level of the hook, those are memoized really well. In all their examples all the objects in the store are really simple objects and each object is maintained seperatly at the top level. But that use case is very derived and unrealistic....

Reply:
That being said, same problem with contexts...